lib/resin-lint.ts

```diff
@@ -0,0 +1,219 @@
+		}
+	}
+
+	return files.map(p => path.join(p));
+};
+
+const lintCoffeeFiles = function(files: string[], config: {}): number {
+	const errorReport = new coffeelint.getErrorReport();
+
+	for (const file of files) {
+		const source = read(file);
+		errorReport.lint(file, source, config);
+	}
+
+	const report = new reporter(errorReport, {
+		colorize: process.stdout.isTTY,
+		quiet: false,
+	});
+
+	report.publish();
+
+	return  errorReport.getExitCode();
+};
+
+const lintTsFiles = function(files: string[], config: {}): number {
+	const parsedConfig = tslint.Configuration.parseConfigFile(config);
+	const linter = new tslint.Linter({
+		fix: false,
+		formatter: 'stylish',
+	});
+
+	for (const file of files) {
+		const source = read(file);
+		linter.lint(file, source, parsedConfig);
+	}
+
+	const errorReport = linter.getResult();
+
+	// Print the linter results
+	console.log(linter.getResult().output);
+
+	return errorReport.errorCount === 0 ? 0 : 1;
+};
+
+const runLint = function(resinLintConfig: ResinLintConfig, paths: string[], config: {}) {
+	let linterExitCode;
+	const scripts = findFiles(resinLintConfig.extensions, paths);
+
+	if (resinLintConfig.lang === 'typescript') {
+		linterExitCode = lintTsFiles(scripts, config);
+	}
+
+	if (resinLintConfig.lang === 'coffeescript') {
+		linterExitCode = lintCoffeeFiles(scripts, config);
+	}
+
+	return process.on('exit', () => process.exit(linterExitCode));
+};
+
+export const lint = function(passedParams) {
+	try {
+		const options = optimist(passedParams)
+			.usage('Usage: resin-lint [options] [...]')
+			.describe('f', 'Specify a linting config file to override resin-lint rules')
+			.describe('p', 'Print default resin-lint linting rules')
+			.describe('i', 'Ignore linting config files in project directory and its parents')
+			.describe('typescript', 'Lint typescript files instead of coffeescript')
+			.boolean('u', 'Run unused import check');
+
+		if ((options.argv._.length < 1) && !options.argv.p) {
+			options.showHelp();
+			process.exit(1);
+		}
+
+		return Promise.try(function() {
+			if (options.argv.u) {
+				return Promise.map(options.argv._, function(dir) {
+					dir = getPackageJsonDir(dir);
+					return Promise.resolve(depcheck(path.resolve('./', dir), {
+						ignoreMatches: [
+							'@types/*', // ignore typescript type declarations
+							'supervisor', // isn't used directly from source
+							'coffee-script', // Gives false positives
+							'coffeescript', // An alias
+							'colors', // Generally imported via colors/safe, which doesn't trigger depcheck
+							'coffeescope2',
+						],
+					}))
+					.get('dependencies')
+					.then(function(deps) {
+						if (deps.length > 0) {
+							console.log(`${deps.length} unused dependencies:`);
+							for (let dep of deps) {
+								console.log(`\t${dep}`);
+							}
+							process.exit(1);
+						}
+						console.log('No unused dependencies!');
+						return console.log();
+					});
+				});
+			}}).then(function() {
```

Comment:
Moving `.then(function() {` to the next line (aligned with `return Promise.try(function() {`) will probably make the indentation better

Reply:
We need prettier here so that we don't have to have these conversations :joy:

Reply:
Yea :)